### DISCUSSION TITLE
depmod was not found in path

### DISCUSSION THREAD

**Discussion by Huhsi** [Freetz-NG/freetz-ng#235] [Allgemein] [answered]
Ich kann (mit der VM von Gismotro) kein Image mehr erstellen.

Wenn ich "make" starte, kommen gleich Fehlermeldungen zu den Programmen execstack und depmod. Das Programm execstack habe ich nachinstalliert - der Fehler ist also behoben. Für depmod habe ich versucht "kmod" nachzuinstallieren. Hier wird dann angezeigt, dass bereits die aktuelle Version installiert ist.

Bei depmod kommt weiterhin die Fehlermeldung "the program depmod was not found in path".

Wo könnte der Fehler liegen?

**Comment by Gismotro** [comment]
Hast du die VM mal im ganzen aktualisiert ?

1.) 32-Bit (Multiarch) aktivieren :

`sudo dpkg --add-architecture i386`
`sudo apt-get -y update`

2.) Pakete updaten:
 `sudo apt-get -y install autoconf automake autopoint bc binutils bison build-essential bzip2 ccache cmake ecj execstack fastjar flex g++ g++-multilib gawk gcc gcc-multilib gettext git imagemagick imagemagick-doc inkscape intltool kmod lib32ncurses5-dev lib32stdc++6 lib32z1-dev libacl1-dev libc6-dev-i386 libcap-dev libglib2.0-dev libncurses5-dev libreadline-dev librsvg2-bin librsvg2-dev libsqlite3-dev libstring-crc32-perl libtool-bin libusb-dev libxml2-dev libzstd-dev libzstd-dev:i386 make ncftp net-tools patch perl pkg-config python rsync ruby ruby1.8 sqlite3 subversion sudo texinfo tofrodos unar unzip wget zlib1g-dev `

**Comment by Huhsi** [comment]
Hatte ich über "update", "-d upgrade" und "-y upgrade" immer aktuell gehalten. Ich habe den Programmen wie von Dir oben angegeben jetzt aber noch einmal ein Update verpasst (waren demnach aber fast alle aktuell).
Der Fehler von depmod besteht aber weiterhin.

**Reply by Gismotro** [reply]
Ich hab die VM soeben neu geladen und nur make gestartet. Beim starten kommen nur die bekannte Fehlermeldung und nach durchlauf der drei Befehle von Oben konnte ich make für eine 7590 starten.

Poste bitte mal deine .config

**Reply by Huhsi** [reply]
Ich bin in Linux leider nicht ganz so fit. Wie kann ich die Config-Einstellungen auslesen und speichern?

Noch eine Frage am Rande:
Wie kann ich Text aus Windows in die Konsole von Linux kopieren (ich habe die drei Befehle von oben fein säuberlich abgeschrieben gehabt)?

**Reply by Gismotro** [reply]
Wo finde ich die .config :
https://freetz-ng.github.io/freetz-ng/wiki/00_FAQ/FAQ.de.html#wo-finde-ich-diese-ber%C3%BChmte-config-datei

Arbeiten per Copy & Paste in der VM : 
https://freetz-ng.github.io/freetz-ng/wiki/10_Beginner/newbie.de.html#putty-starten

**Comment by fda77** [comment, marked as answer]
Am besten immer https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES.md verlinken, die Pakete können sich ändern! In FOS 7.2x kam depmod hinzu, in 7.25 execstack und rsync

Das ganze abzutippen ist die schlechteste Lösung. Am besten mit "Putty" per SSH zur VM verbinden. Deren IP findet man mit "ifconfig" oder im Router

**Reply by Huhsi** [reply]
Mit Putty war einer sehr guter Tipp. Damit ging es deutlich einfacher. Aber der Fehler besteht weiterhin.
Ich habe die VM auch noch einmal neu aufgesetzt - ohne Veränderung.

**Reply by Huhsi** [reply]
Alles zurück. Ich habe hinterher  (nach den drei o.g. Befehlen) noch einmal die Update  gestartet (update, -d upgrade, -y upgrade). Jetzt funktioniert läuft es... 

Vielen Dank für die Unterstützung.

**Comment by josch09** [comment]
Also ich hatte das gleiche Problem.

Mein Ubuntu (16.04.7) läuft in einem LXC container. Das funzt wunderbar bei freetz, freetz-ng, openwrt/LEDE oder auch openatv - eine vollständige VM als Buildhost ist nicht notwendig, wenn man wie ich daheim eh nur Linux/BSD am Laufen hat (und kein Ubuntu oder Frickelbuden-Fedora Fan ist). In diesem LXC Container bau ich seit Jahren meine ganzen Home-Entertainment Linuxe.

In meiner Umgebung war es ein reines PATH Problem (denn installiert war es).

- depmod liegt in /sbin (und depmod selbst ist ein symlink auf /bin/kmod - so wie es sein soll)
- ein normaler Linux-User (ohne sudo/doas whatever), der jetzt die 7.2er maken will, hat "/sbin" im Regelfall nicht im PATH environment, weil nach alter UNIX/Linux-Tradition dort eigentlich nur Programme liegen, mit denen nur der Admin/Superuser (root) was anfangen kann/darf... und prallt dann mit der Fehlermeldung ab.

- also "/sbin" der PATH env variable hinzufügen und schon gehts... oder die make-Umgebung ruft gleich direkt /sbin/depmod auf (wär auch'ne Moglichkeit)

**Reply by fda77** [reply]
Die Datei muss nicht zwingend als /sbin/depmod gespeichert sein. Wenn sie sonstwo liegt wird sie mit festen Pfad auf keinen Fall gefunden.
Versuch mal https://github.com/Freetz-NG/freetz-ng/blob/master/fwmod#L105 nach `PATH="${PATH_DIR}:${PATH}:/sbin"` zu ändern.
ODER
einen Link mir `ln -sf /sbin/depmod tools/path/` anlegen

**Reply by fda77** [reply]
Bei Fedora ist es übrigens /usr/sbin/depmod und im Pfad enthalten (/sbin ist ein Link)
